From 1b4dce6f843d50fbc9b0fa8f86527788fb0c29d7 Mon Sep 17 00:00:00 2001
From: Adel Gadllah <adel.gadllah@gmail.com>
Date: Thu, 29 Sep 2011 13:16:33 +0000
Subject: meta-window-group: Plug memory leak

We have to free the list returned by clutter_actor_get_effects() .

https://bugzilla.gnome.org/show_bug.cgi?id=660464
---
diff --git a/src/compositor/meta-window-group.c b/src/compositor/meta-window-group.c
index 107e7ba..8889503 100644
--- a/src/compositor/meta-window-group.c
+++ b/src/compositor/meta-window-group.c
@@ -108,7 +108,7 @@ meta_window_group_paint (ClutterActor *actor)
   cairo_region_t *unredirected_window_region = NULL;
   ClutterActor *stage;
   cairo_rectangle_int_t visible_rect, unredirected_rect;
-  GList *children, *l;
+  GList *children, *l, *effects;
 
   MetaWindowGroup *window_group = META_WINDOW_GROUP (actor);
   MetaCompScreen *info = meta_screen_get_compositor_data (window_group->screen);
@@ -161,8 +161,11 @@ meta_window_group_paint (ClutterActor *actor)
        * as well for the same reason, but omitted for simplicity in the
        * hopes that no-one will do that.
        */
-      if (clutter_actor_get_effects (l->data) != NULL)
-        continue;
+      if ((effects = clutter_actor_get_effects (l->data)) != NULL)
+        {
+          g_list_free (effects);
+          continue;
+        }
 
       if (META_IS_WINDOW_ACTOR (l->data))
         {
--
cgit v0.9.0.2
